### ISSUE TITLE
Issue with DTH

### ISSUE THREAD

**Issue by rlg6767** [GaryMilne/SmartThingsUnifiedPresence#1]
First of all,  many thanks for creating this app!!   Have built the raspberry pi side of things and all seems to be working.   However,  when I go to create the device type handler in Smartthings IDE I get:
Org.hibernate.exception.GenericJDBCException: Could not execute JDBC batch update
and it fails to create the DTH.   Any ideas?
Many thanks,
Richard.

**Comment by GaryMilne** [comment]
Yes, some people get the error and it is related to the presence of Icons in the code.  Don't know why some people get it and others don't.  So you will find all of the icons in two functions near the top of the code.  Simple pick a letter and replace the icon with that letter until there are no icons left.

Then it should publish O.K.  You could try at that point putting the Icons back in and publishing again.

**Comment by rlg6767** [comment]
Yes,  this has now fixed this issue,  many thanks.   However I have now noticed a separate issue whereby within the Smartthings app all 3 sensors (2 x BT,  1 x IP) are showing offline, when I disable wireless and BT on the device,  however the master presence stays as "Home".   Have you encountered this before?

**Comment by GaryMilne** [comment]
For a device to go from in to out it must miss successive check-ins. This prevents accidental triggering of a change of state due to network conditions. I think it is three attempts.  You will find the number prominently near the top of the python script along with the polling frequency.

**Comment by rlg6767** [comment]
I set the poll interval to 1 minute and half an hour later the master status still hasn't changed despite the individual sensors all showing "away" within one minute of turning BT and wireless off on the mobile.

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________
